src/main/java/Duke.java

```diff
@@ -1,10 +1,142 @@
+            String[] userCommands = userInput.split(" ", 2);
+            if (userCommands[0].equalsIgnoreCase("bye")) {
+                printByeMessage();
+                break;
+            } else if (userCommands[0].equalsIgnoreCase("list")) {
+                printList(Tasks, taskCounter);
+            } else if (userCommands[0].equalsIgnoreCase("done")) {
+                printDone(userCommands, Tasks);
+            } else if(userCommands[0].equalsIgnoreCase("todo")) {
+                addTodo(Tasks, taskCounter, userCommands);
+                taskCounter ++;
+            } else if(userCommands[0].equalsIgnoreCase("event")) {
+                addEvent(Tasks, taskCounter, userCommands[1]);
+                taskCounter++;
+            } else if(userCommands[0].equalsIgnoreCase("deadline")) {
+                addDeadline(Tasks, taskCounter, userCommands[1]);
+                taskCounter++;
+            } else {
+                addTask(Tasks, taskCounter, userInput);
+                taskCounter++;
+            }
+            System.out.println(DIVIDER);
+        }
+    }
+
+    private static void addTask(Task[] Tasks, int taskCounter, String userInput) {
+        String buffer = userInput;
+        Tasks[taskCounter] = new Task(buffer.trim());
+        System.out.println("    added: " + Tasks[taskCounter].description);
+    }
+
+    private static void addEvent(Task[] Tasks, int taskCounter, String userCommand) {
+        try {
+            String[] buffer = userCommand.split("/at", 2);
+            Tasks[taskCounter] = new Event(buffer[0].trim(), buffer[1].trim());
+            printTask(Tasks[taskCounter], taskCounter);
+        } catch (Exception e){
+            System.out.println("    Error: Insufficient detail");
+        }
+    }
+
+    private static void addDeadline(Task[] Tasks, int taskCounter, String userCommand) {
+        try {
+            String[] buffer = userCommand.split("/by", 2);
+            Tasks[taskCounter] = new DeadLine(buffer[0].trim(), buffer[1].trim());
+            printTask(Tasks[taskCounter], taskCounter);
+        } catch(Exception e) {
+            System.out.println("    Error: Insufficient detail");
+        }
+    }
+
+    private static void addTodo(Task[] Tasks, int taskCounter, String[] userCommands) {
+        try {
+            Tasks[taskCounter] = new Todo(userCommands[1]);
+            printTask(Tasks[taskCounter], taskCounter);
+        } catch(Exception e) {
+            System.out.println("    Error: Insufficient detail");
+        }
+    }
+
+    private static void printDone(String[] input, Task[] Tasks) {
+        try {
+            int num = Integer.parseInt(input[1]) - 1;
+            Tasks[num].taskDone();
+            System.out.println("    Nice! I've marked this task as done:");
+            System.out.println("      " + Tasks[num]);
+        }catch(NullPointerException e) {
+            System.out.println("    Error: Given Index is out of bound");
+        }catch(Exception e) {
```

Comment:
Should there be a spacing between the '}' and catch as seen in the other try-catch sections?

Reply:
thanks for telling me 😄